esp_hosted_ng/host/sdio/esp_sdio.c

```diff
@@ -728,6 +728,18 @@ static int esp_probe(struct sdio_func *func,
 		return -ENOMEM;
 	}
 
+	if (sdio_context.sdio_clk_mhz){
```

Comment:
Can we assign `sdio_context.sdio_clk_mhz` to some default value, say 50, in case the module param is not passed?

Ideally, 
```sh
	if (!sdio_context.sdio_clk_mhz)
		sdio_context.sdio_clk_mhz = SDIO_INITIAL_CLK_MHZ;

	/* TODO: move these declarations in start of the function */
	struct mmc_host *host = func->card->host;
	u32 hz = sdio_context.sdio_clk_mhz * 1000000;

	/* Expantion of mmc_set_clock that isnt exported */
	if (hz < host->f_min)
		hz = host->f_min;
	if (hz > host->f_max)
		hz = host->f_max;
	host->ios.clock = hz;
	host->ops->set_ios(host, &host->ios);
```

Reply:
As of right now it doesn't set it to anything so what value is default in the device tree is what is used for the raspberry pi this is 50mhz which is far too fast for jumper cables. I can do it like that but prioritizing the device tree makes the most sense

Reply:
Thank you @Mr-Bossman for all your efforts.

That (DT) would always be the problem. The device tree is very dependent on SoC. Most commonly, we face issues that user do not know how to define Device Tree correctly or someone else has done it for them.

But then `sdio_context.sdio_clk_mhz` **has to be initialized** with the value from device tree. Correct? I think we can ask user to do in the documentation by passing the module param.
In case module param not provided, At least some non-zero value (considering all common SDIO devices) should at least be there, I think.

In case, user want to override with some other freq than that of mentioned in DT, can do using module param.

Agree?

Reply:
I'm not sure I'm understanding. As of right now passing nothing will not touch the frequency. It will act as if this patch hasn't existed.